src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -703,4 +707,479 @@ private async Task<HttpResponseMessage> GetVersion(Guid dataSetId, string dataSe
+                .WithStatusPublished()
+                .WithDataSetId(dataSet.Id)
+                .WithFilterMetas(() => filterMetas)
+                .WithLocationMetas(() => locationMetas)
+                .WithIndicatorMetas(() =>
+                    DataFixture
+                    .DefaultIndicatorMeta()
+                    .GenerateList(3)
+                )
+                .WithTimePeriodMetas(() =>
+                    DataFixture
+                    .DefaultTimePeriodMeta()
+                    .GenerateList(3)
+                )
+                .WithMetaSummary();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSetVersions.Add(dataSetVersion));
+
+            var response = await GetMeta(dataSet.Id, dataSetVersion.Version);
+
+            var content = response.AssertOk<DataSetMetaViewModel>(useSystemJson: true);
+
+            Assert.NotNull(content);
+
+            foreach (var filterMetaViewModel in content.Filters)
+            {
+                var filterMeta = Assert.Single(dataSetVersion.FilterMetas, fm => fm.PublicId == filterMetaViewModel.Id);
+                Assert.Equal(filterMeta.Hint, filterMetaViewModel.Hint);
+                Assert.Equal(filterMeta.Label, filterMetaViewModel.Label);
+
+                var allFilterMetaLinks = filterMeta.Options.SelectMany(fom => fom.MetaLinks);
+
+                foreach (var filterOptionMetaViewModel in filterMetaViewModel.Options)
+                {
+                    var filterOptionMetaLink = Assert.Single(
+                        allFilterMetaLinks,
+                        foml => SqidEncoder.Encode(foml.PublicId) == filterOptionMetaViewModel.Id);
+
+                    var filterOptionMeta = Assert.Single(
+                        filterMeta.Options,
+                        fom => fom.Id == filterOptionMetaLink.OptionId);
+
+                    Assert.Equal(filterOptionMeta.Label, filterOptionMetaViewModel.Label);
+                    Assert.Equal(filterOptionMeta.IsAggregate, filterOptionMetaViewModel.IsAggregate);
+                }
+            }
+
+            foreach (var locationMetaViewModel in content.Locations)
+            {
+                var locationMeta = Assert.Single(
+                    dataSetVersion.LocationMetas, 
+                    fm => fm.Level == locationMetaViewModel.Level);
+
+                var allLocationMetaLinks = locationMeta.Options.SelectMany(fom => fom.MetaLinks);
+
+                foreach (var locationOptionMetaViewModel in locationMetaViewModel.Options)
+                {
+                    var locationOptionMetaLink = Assert.Single(
+                        allLocationMetaLinks,
+                        foml => SqidEncoder.Encode(foml.PublicId) == locationOptionMetaViewModel.Id);
+
+                    var locationOptionMeta = Assert.Single(
+                        locationMeta.Options,
+                        fom => fom.Id == locationOptionMetaLink.OptionId);
```

Comment:
This isn't the right abbreviation, but personally, I'd just call this `o` anyway (similarly with the filter option meta above).